samples/weak/basic.jakt

```diff
@@ -0,0 +1,19 @@
+class Foo {
+    public function hello(this) => "friends"
+}
+
+function main() {
+    let mutable weak_foo: weak Foo = None
```

Comment:
Per earlier discussion it was agreed that this would use `weak Foo?`, did that change?

Reply:
That didn't change. Hence "Work towards #170" instead of "Fixes #170" :^)